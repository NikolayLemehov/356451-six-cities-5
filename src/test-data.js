// import {upperCaseFirst} from "./utils";

// const currentCityOffers = [
//   {
//     id: `1`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 226,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
//   {
//     id: `4`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 126,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
//   {
//     id: `7`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 256,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
// ];
export const testInitialState = {
  COMMON: {
    offers: [],
    nearOffers: [],
    changedBookmarkOffer: {},
    pageOffer: {},
    bookmarkOffers: [],
    authInfo: {},
    cities: [`City1`, `City2`, `City3`, `City4`, `City5`, `City6`],
    currentCityName: `City1`,
    currentSortType: `Popular`,
    overOfferId: ``,
    reviews: [],
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
  },
};
