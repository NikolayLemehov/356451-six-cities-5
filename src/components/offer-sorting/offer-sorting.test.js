import React from "react";
import renderer from "react-test-renderer";
import OfferSorting from "./offer-sorting";
import configureMockStore from "redux-mock-store";
import {testInitialState} from "../../test-data";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

const mockStore = configureMockStore();
const store = mockStore(testInitialState);
describe(`OfferSorting should render correctly`, () => {
  it(`OfferSorting should render correctly when the menu is closed`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <OfferSorting/>
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it(`OfferSorting should render correctly when the menu is open`, () => {
  const tree = mount(
      <Provider store={store}>
        <MemoryRouter>
          <OfferSorting/>
        </MemoryRouter>
      </Provider>
  );
  const menuButton = tree.find(`.places__sorting-type`);
  menuButton.simulate(`click`);
  expect(tree.debug()).toMatchSnapshot();
  tree.unmount();
});
