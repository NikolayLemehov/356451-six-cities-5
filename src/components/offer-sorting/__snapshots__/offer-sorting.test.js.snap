// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferSorting should render correctly OfferSorting should render correctly when the menu is closed 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`OfferSorting should render correctly when the menu is open 1`] = `
"<Provider store={{...}}>
  <MemoryRouter>
    <Router history={{...}}>
      <OfferSorting currentSortType=\\"Popular\\">
        <form className=\\"places__sorting\\" action=\\"#\\" method=\\"get\\">
          <span className=\\"places__sorting-caption\\">
            Sort by
          </span>
          <span className=\\"places__sorting-type\\" tabIndex=\\"0\\" onClick={[Function: clickMenuHandle]}>
            Popular
            <svg className=\\"places__sorting-arrow\\" width=\\"7\\" height=\\"4\\">
              <use xlinkHref=\\"#icon-arrow-select\\" />
            </svg>
          </span>
          <ul className=\\"places__options places__options--custom places__options--opened\\">
            <Connect(OfferSortingItem) type=\\"Popular\\" onCloseMenu={[Function: onCloseMenu]}>
              <OfferSortingItem type=\\"Popular\\" onCloseMenu={[Function: onCloseMenu]} currentSortType=\\"Popular\\" onChangeSortedType={[Function: onChangeSortedType]}>
                <li className=\\"places__option places__option--active\\" tabIndex=\\"0\\" onClick={[Function: onClick]}>
                  Popular
                </li>
              </OfferSortingItem>
            </Connect(OfferSortingItem)>
            <Connect(OfferSortingItem) type=\\"Price: low to high\\" onCloseMenu={[Function: onCloseMenu]}>
              <OfferSortingItem type=\\"Price: low to high\\" onCloseMenu={[Function: onCloseMenu]} currentSortType=\\"Popular\\" onChangeSortedType={[Function: onChangeSortedType]}>
                <li className=\\"places__option\\" tabIndex=\\"0\\" onClick={[Function: onClick]}>
                  Price: low to high
                </li>
              </OfferSortingItem>
            </Connect(OfferSortingItem)>
            <Connect(OfferSortingItem) type=\\"Price: high to low\\" onCloseMenu={[Function: onCloseMenu]}>
              <OfferSortingItem type=\\"Price: high to low\\" onCloseMenu={[Function: onCloseMenu]} currentSortType=\\"Popular\\" onChangeSortedType={[Function: onChangeSortedType]}>
                <li className=\\"places__option\\" tabIndex=\\"0\\" onClick={[Function: onClick]}>
                  Price: high to low
                </li>
              </OfferSortingItem>
            </Connect(OfferSortingItem)>
            <Connect(OfferSortingItem) type=\\"Top rated first\\" onCloseMenu={[Function: onCloseMenu]}>
              <OfferSortingItem type=\\"Top rated first\\" onCloseMenu={[Function: onCloseMenu]} currentSortType=\\"Popular\\" onChangeSortedType={[Function: onChangeSortedType]}>
                <li className=\\"places__option\\" tabIndex=\\"0\\" onClick={[Function: onClick]}>
                  Top rated first
                </li>
              </OfferSortingItem>
            </Connect(OfferSortingItem)>
          </ul>
        </form>
      </OfferSorting>
    </Router>
  </MemoryRouter>
</Provider>"
`;
